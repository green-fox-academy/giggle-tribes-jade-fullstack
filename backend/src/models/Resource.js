import {
  insertResourceForKingdom,
  getResourceForKingdom,
  updateResourceForKingdom,
} from '../repos/resource';

export class Resource {
  kingdomID;
  type;
  amount;
  generation;
  updatedAt;

  constructor(kingdomID, type, amount) {
    this.kingdom = resource.results[0].kingdom_id;
    this.type = type;
    this.amount = amount;
    this.generation = 1;
    this.updatedAt = updatedAt;
  }
  getResource(kingdom, type, amount, generation) {
    return {
      type: this.type,
      amount: this.amount,
      generation: this.generation,
      updatedAt: this.updatedAt,
    };
  }
}
