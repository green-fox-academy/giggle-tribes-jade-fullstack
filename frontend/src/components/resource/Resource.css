.resources {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr;
  background-color: rgba(201, 204, 205, 0.8);
  height: 175px;
  width: 300px;
}
img {
  height: 40px;
  width: 40px;
}

img.resource {
  height: 70px;
  width: 70px;
}

.resource.farm {
  grid-area: 1 / 1 /2 /1;
  justify-self: center;
  align-self: center;
}
.resource.food {
  grid-area: 1 / 2 /1 /4;
  justify-content: center;
  align-items: flex-start;
  display: flex;
  flex-direction: column;
}
.resource.mine {
  grid-area: 2 / 1 /2 /1;
  justify-self: center;
  align-self: center;
}
.resource.gold {
  grid-area: 2 / 2 /2 /4;
  align-self: center;
  align-items: flex-start;
  display: flex;
  flex-direction: column;
}
.resource div {
  display: flex;
}
img.food,
img.gold {
  align-self: center;
  margin-left: 10px;
}
h1 {
  margin: 0px 0px 0px 0px;
  font-size: 40px;
  color: #5c5e5e;
}
p {
  margin: 0px 0px 0px 0px;
}
