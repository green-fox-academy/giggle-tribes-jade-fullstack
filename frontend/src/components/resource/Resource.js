import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import './Resource.css';
import Farm from '../../assets/buildings/farm.svg';
import Mine from '../../assets/buildings/mine.svg';
import Food from '../../assets/sources/FoodIcon.svg';
import Gold from '../../assets/sources/GoldIcon.svg';

export default function Resource({ kingdomID }) {
  const [foodAmount, setFoodAmount] = useState(null);
  const [goldAmount, setGoldAmount] = useState(null);

  const getResources = async kingdomID => {
    await fetch(`http://localhost:5000/api/kingdoms/${kingdomID}/resource`, {
      headers: {
        'Content-Type': 'application/json',
        TRIBES_TOKEN: localStorage.getItem('TRIBES_TOKEN'),
      },
    })
      .then(res => res.json())
      .then(response => response.resources)
      .then(resources =>
        resources.map(async resource => {
          if (resource.type === 'food') {
            setFoodAmount(resource.amount);
          } else {
            setGoldAmount(resource.amount);
          }
        })
      )
      .catch(console.log);
  };

  useEffect(() => {
    getResources(kingdomID);
  }, [kingdomID]);

  return (
    <div className="resources">
      <img className="resource farm" src={Farm} alt="farm icon" />
      <div className="resource food">
        <div>
          <h1>{foodAmount}</h1>
          <img className="food" src={Food} alt="food icon" />
        </div>
        <p>{} / minute</p>
      </div>
      <img className="resource mine" src={Mine} alt="mine icon" />
      <div className="resource gold">
        <div>
          <h1>{goldAmount}</h1>
          <img className="gold" src={Gold} alt="gold icon" />
        </div>
        <p>{} / minute</p>
      </div>
    </div>
  );
}

Resource.propTypes = {
  kingdomID: PropTypes.any.isRequired,
};
