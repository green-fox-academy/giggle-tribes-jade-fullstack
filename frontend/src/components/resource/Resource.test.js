import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';

import Resource from './Resource';

global.fetch = jest.fn(async () => {
  return await Promise.resolve({
    json: () =>
      Promise.resolve({
        resources: [
          {
            type: 'food',
            amount: 6824,
            generation: 1,
            updatedAt: '2020-07-08T18:32:01.000Z',
          },
          {
            type: 'gold',
            amount: 6824,
            generation: 1,
            updatedAt: '2020-07-08T18:32:01.000Z',
          },
        ],
      }),
  });
});

let container;

beforeEach(() => {
  fetch.mockClear();
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

let foodAmount;
let goldAmount;

function setFoodAmount(amount) {
  foodAmount = amount;
  console.log(foodAmount);
}
function setGoldAmount(amount) {
  goldAmount = amount;
}

it('matches snapeshot', async () => {
  await act(async () => {
    await render(
      <Resource
        kingdomID={1}
        foodAmount={foodAmount}
        setFoodAmount={setFoodAmount}
        goldAmount={goldAmount}
        setGoldAmount={setGoldAmount}
      />,
      container
    );
  });

  expect(container.innerHTML).toMatchSnapshot();
});
