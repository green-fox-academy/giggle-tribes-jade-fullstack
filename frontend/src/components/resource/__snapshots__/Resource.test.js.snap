// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapeshot 1`] = `"<div class=\\"resources\\"><img class=\\"resource farm\\" src=\\"farm.svg\\" alt=\\"farm icon\\"><div class=\\"resource food\\"><div><h1>6824</h1><img class=\\"food\\" src=\\"FoodIcon.svg\\" alt=\\"food icon\\"></div><p> / minute</p></div><img class=\\"resource mine\\" src=\\"mine.svg\\" alt=\\"mine icon\\"><div class=\\"resource gold\\"><div><h1>6824</h1><img class=\\"gold\\" src=\\"GoldIcon.svg\\" alt=\\"gold icon\\"></div><p> / minute</p></div></div>"`;
